Trakt/Extensions.cs

```diff
@@ -539,4 +550,39 @@
 
         return false;
     }
+
+    /// <summary>
+    /// Checks if a <see cref="Episode"/> matches a <see cref="TraktEpisodeWatchedHistory"/>.
+    /// </summary>
+    /// <param name="item">The <see cref="Episode"/>.</param>
+    /// <param name="episodeHistory">The <see cref="TraktEpisodeWatchedHistory"/>.</param>
+    /// <returns><see cref="bool"/> indicating if the <see cref="Episode"/> matches a <see cref="TraktEpisodeWatchedHistory"/>.</returns>
+    public static bool IsMatch(Episode item, TraktEpisodeWatchedHistory episodeHistory)
+    {
+        // Match by provider id's
+        if (IsMatch(item, episodeHistory.Episode))
+        {
+            return true;
+        }
+
+        // Match by show, season and episode number if there isn't any provider id in common
+        // If there was a common provider id between the item and the trakt episode (f.e. both have tvdb id), you shouldn't check anymore by season/number
+        if (HasAnyProviderTvIdInCommon(item, episodeHistory.Episode)
+            && IsMatch(item.Series, episodeHistory.Show)
+            && item.GetSeasonNumber() == episodeHistory.Episode.Season
+            && item.ContainsEpisodeNumber(episodeHistory.Episode.Number))
+        {
+            return true;
+        }
+
+        return false;
+    }
+
+    private static bool HasAnyProviderTvIdInCommon(Episode item, TraktEpisode traktEpisode)
+    {
+        return (item.HasProviderId(MetadataProvider.Tvdb) && traktEpisode.Ids.Tvdb != null)
+            || (item.HasProviderId(MetadataProvider.Imdb) && traktEpisode.Ids.Imdb != null)
+            || (item.HasProviderId(MetadataProvider.Tmdb) && traktEpisode.Ids.Tmdb != null)
+            || (item.HasProviderId(MetadataProvider.TvRage) && traktEpisode.Ids.Tvrage != null);
```

Comment:
I don't think there is any way to implement this otherwise.
We need to check if the item and the traktEpisode have at least 1 of the 4 provider id's in common (defined and not null).